runners/flink/src/main/java/org/apache/beam/runners/flink/FlinkPortableRunnerResult.java

```diff
@@ -44,13 +50,44 @@ public JobApi.MetricResults portableMetrics() throws UnsupportedOperationExcepti
         .build();
   }
 
-  static class Detached extends FlinkDetachedRunnerResult implements PortablePipelineResult {
+  static class Detached implements PortablePipelineResult {
+
+    Detached() {
+      super();
+    }
 
     @Override
     public JobApi.MetricResults portableMetrics() throws UnsupportedOperationException {
       LOG.warn(
           "Collecting monitoring infos is not implemented yet in Flink portable runner (detached mode).");
       return JobApi.MetricResults.newBuilder().build();
     }
+
+    @Override
+    public @UnknownKeyFor @NonNull @Initialized State getState() {
```

Comment:
These "@UnknownKeyFor @NonNull @Initialized" are not necessary. These are added by default to compiled .class. IDE may auto propagated them which is slightly annoying.

Reply:
Fixed